### PR TITLE
av-set: expose fault domain count on convert

### LINKED ISSUE #1800: VM Creation - SkuNotAvailable when creating VM in region
In the CLI, I tried creating a VM in 'West US' in my subscription but kept on getting a `ResourceDeploymentFailure` with code `SkuNotAvailable`.

```
At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details. {
  "status": "Failed",
  "error": {
    "code": "ResourceDeploymentFailure",
    "message": "The resource operation completed with terminal provisioning state 'Failed'.",
    "details": [
      {
        "code": "DeploymentFailed",
        "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details.",
        "details": [
          {
            "code": "Conflict",
            "message": "{\r\n  \"error\": {\r\n    \"code\": \"SkuNotAvailable\",\r\n    \"message\": \"The requested tier for resource '/subscriptions/.../resourceGroups/commute-dev/providers/Microsoft.Compute/virtualMachines/commutedev1' is currently not available in location 'westus' for subscription '...'. Please try another tier or deploy to a different location.\"\r\n  }\r\n}"
          }
        ]
      }
    ]
  }
}
```

I tried many VM sizes and kept getting the same result.

I then tried in the portal and get the error message "Your subscription doesn't support virtual machine creation in West US."

screen shot 2017-01-19 at 11 26 43 am

The CLI doesn't show a list of supported regions and there doesn't appear to be a way to get this.

Can we use the same logic/API calls as the portal?

### LINKED ISSUE #3790: VM: availability-set convert command should expose args to set FD count
Managed disk based AV set has less FD count than un-managed version, hence this arg makes sense. 
UD count can not be tweaked.

### PR DESCRIPTION
**Pull request by yugangw-msft**
Fix #3790, partial address #1800  by exposing `az vm list-skus`
- [x] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by yugangw-msft** [comment]
New command:
```bash
Command
    az vm list-skus: Get compute related resource skus, including vm size, fault domain count for an
    availability set, etc.
        This command incorporates subscription level restriction, hence offers the most accurate
        information.

Arguments
    --location -l: Location. You can configure the default location using `az configure --defaults
                   location=<location>`.
```

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -96,6 +96,7 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):  # pylin
                       help='Update Domain count. If unspecified, server picks the most optimal number like 5. For the latest defaults see https://docs.microsoft.com/en-us/rest/api/compute/availabilitysets/availabilitysets-create')
 register_cli_argument('vm availability-set create', 'platform_fault_domain_count', type=int, help='Fault Domain count.')
 register_cli_argument('vm availability-set create', 'validate', help='Generate and validate the ARM template without creating any resources.', action='store_true')
+register_cli_argument('vm availability-set convert', 'platform_fault_domain_count', type=int, help="New Fault Domain count if provided. Use 'az vm list-skus -l <location>' to check out available count")
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: You should probably use the "populator-command" feature in `help.py` to take advantage of the standard formatting here.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py [hunk 2]
```diff
@@ -1528,12 +1528,16 @@ def vmss_set(resource_group_name, name, no_wait=False, **kwargs):
         resource_group_name, name, raw=no_wait, **kwargs)
 
 
-def convert_av_set_to_managed_disk(resource_group_name, availability_set_name):
+def convert_av_set_to_managed_disk(resource_group_name, availability_set_name, platform_fault_domain_count=None):
     av_set = availset_get(resource_group_name, availability_set_name)
     if av_set.sku.name != 'Aligned':
         av_set.sku.name = 'Aligned'
+        if platform_fault_domain_count:
+            av_set.platform_fault_domain_count = platform_fault_domain_count
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py]: Now that we have that API available, should we not retrieve the domain count and use the max supported value here?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py [hunk 3]
```diff
@@ -1528,12 +1528,16 @@ def vmss_set(resource_group_name, name, no_wait=False, **kwargs):
...
         return availset_set(resource_group_name=resource_group_name, name=availability_set_name,
                             parameters=av_set)
+    else:
+        logger.warning('The availability set is already managed disk based, hence skipping...')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py]: Rec: "Availability set {name} is already configured for managed disks."

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py (hunk 1 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: per our conversation, we don't need to expose this argument, so i have removed it

**Pull request merged**